File: SiriusJavaClient-Web/src/main/java/org/sirius/client/web/classes/WebCheckBox.java
/**
 * 
 */
package org.sirius.client.web.classes;

import java.rmi.RemoteException;

/**
 * @author Myk Kolisnyk
 * 
 */
public class WebCheckBox extends WebRadioButton {

    /**
     * @param parent
     * @param locator
     */
    public WebCheckBox(final Frame parent, final String locator) {
        super(parent, locator);
    }

    /**
     * @param parent
     * @param parentElement
     * @param locator
     */
    public WebCheckBox(final Frame parent, final String parentElement,
            final String locator) {
        super(parent, parentElement, locator);
    }

    public void uncheck() throws RemoteException, Exception {
        if (isChecked()) {
            click();
        }
    }

}

File: sirius-ruby-client-web/lib/sirius/web/classes/web_checkbox.rb
module Sirius
  module Client
    module Web
      module Classes
        class WebCheckBox < WebRadioButton
          def initialize(parent, locator, parentElement = nil)
            super(parent, locator, parentElement)
          end

          def uncheck
            click if checked?
          end
        end
      end
    end
  end
end